cmd/kflex/adopt/adopt.go

```diff
@@ -0,0 +1,224 @@
+/*
+Copyright 2024 The KubeStellar Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package adopt
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"sync"
+
+	"path/filepath"
+
+	homedir "github.com/mitchellh/go-homedir"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/clientcmd/api"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	tenancyv1alpha1 "github.com/kubestellar/kubeflex/api/v1alpha1"
+	"github.com/kubestellar/kubeflex/cmd/kflex/common"
+	cont "github.com/kubestellar/kubeflex/cmd/kflex/ctx"
+	kfclient "github.com/kubestellar/kubeflex/pkg/client"
+	"github.com/kubestellar/kubeflex/pkg/util"
+)
+
+type CPAdopt struct {
+	common.CP
+	AdoptedKubeconfig             string
+	AdoptedContext                string
+	AdoptedURLOverride            string
+	AdoptedTokenExpirationSeconds int
+	SkipURLOverride               bool
+}
+
+// Adopt a control plane from another cluster
+func (c *CPAdopt) Adopt(hook string, hookVars []string, chattyStatus bool) {
+	done := make(chan bool)
+	var wg sync.WaitGroup
+	cx := cont.CPCtx{}
+	cx.Context(chattyStatus, false, false, false)
+
+	controlPlaneType := tenancyv1alpha1.ControlPlaneTypeExternal
+	util.PrintStatus(fmt.Sprintf("Adopting control plane %s of type %s ...", c.Name, controlPlaneType), done, &wg, chattyStatus)
```

Comment:
Did you know that a Cobra `Command` has methods like `Context`, `ErrOrStderr`, `OutOrStdout`?